File: leetcode/golang/1137.泰波那契数列.go
func tribonacci(n int) int {
    var d []int = []int{0, 1, 1};
    if n < 3 {
        return d[n];
    }
    i := 3;
    for {
        if i == n + 1{
            break;
        }
        num := d[2] + d[1] + d[0];
        if i == n {
            return num;
        }
        d = append(d[1:], num);
        i += 1;
    }
    return d[n];
}

File: leetcode/python/1137.泰波那契数列.py
class Solution:
    def tribonacci(self, n: int) -> int:
        d = [0, 1, 1]
        for i in range(3, n + 1):
            num = sum(d[-3:])
            d.append(num)
            if i == n:
                return num
        return d[n]